src/tests/system/docs/writing-tests.rst

```diff
@@ -0,0 +1,408 @@
+Writing new test
+################
+
+The tests are written using the `pytest`_ framework, `pytest-mh`_ plugin and
+SSSD specific extensions that implements related :mod:`~lib.sssd.hosts` and
+:mod:`~lib.sssd.roles`.
+
+This article covers the basic knowledge required to write a new test. After you
+finish it, make sure to go through the :doc:`course` and :doc:`guides/index`.
+You will also benefit from reading the :doc:`api`.
+
+.. seealso::
+
+    It is highly recommended to read `pytest`_ and `pytest-mh`_ documentation so
+    you can write your tests with all features and tools that are available.
+
+.. _pytest: https://docs.pytest.org
+.. _pytest-mh: https://pytest-mh.readthedocs.io
+
+Using the topology marker
+*************************
+
+Each test that requires access to hosts defined in multihost configuration must
+be marked with a ``topology`` marker. This marker provides information about the
+topology that is required to run the test and defines fixture mapping between a
+short fixture name and a host from the multihost configuration (this is
+explained later in `Deep dive into multihost fixtures`_).
+
+The marker is used as:
+
+.. code-block:: python
+
+    import pytest
+
+
+    @pytest.mark.topology(name, topology, fixtures ...)
+    def test_example():
+        assert True
```

Comment:
There should be a test example with docstring describing setup, stepe and expected results.

Reply:
Reference to betelgeuse would also make a sense.